src/app.js

```diff
@@ -228,7 +234,23 @@ export function app() {
 		}
 
 		switch (message.type) {
-			case 'PUBLISH':
+			case 'CONNACK': {
+				debug('CONNECTED %o', message);
+
+				handleConnect(event);
+
+				const queueLength = flushQueue(token);
```

Comment:
flushing the queue needs to be done before we call the `connect` lifecycle method, otherwise the tests will freak out.

Reply:
an alternative is have another lifecycle step, to make things more clear (possibly more complex, I guess):

`connect`
`drain` / `flush`
`spawn`
`destroy` ??

Reply:
don't think it should be in the lifecycle